### PR TITLE
feat: add draggable file upload component

### PR DESCRIPTION
**Pull request by 4nalog**
# TL;DR

This PR implements a generic file upload component with support for dragging/dropping to upload.

## Demo

https://user-images.githubusercontent.com/88684372/220675769-21b84ea5-dd40-45d9-8bc5-60a5d346135e.mov

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

This PR:
- adds [react-dropzone](https://react-dropzone.js.org/#section-extending-dropzone) as dependency
- adds a `DraggableFileUpload` component to support uploading files with drag and drop functionality

## Tracking Issue

[#3317](https://github.com/flyteorg/flyte/issues/3317)

## Follow-up issue

_https://github.com/flyteorg/flyte/issues/3317_

### CODE REVIEW

File: packages/console/src/components/common/FileUpload/FileUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}));
+
+interface Props {
+  files: File[];
+  setFiles: React.Dispatch<React.SetStateAction<File[]>>;
+  options?: DropzoneProps;
+  helpText?: React.ReactNode;
+}
+
+export function FileUpload({ files, setFiles, options, helpText }: Props) {
```
[the remark this answers is not in the document]
Comment (meehawk) [reply packages/console/src/components/common/FileUpload/FileUpload.tsx:39]: Yes, we would need to send a `accepted` prop this is supported by the react-dropzone library.

The help text below can be passed via `helpText` prop

[the remark this answers is not in the document]
Comment (4nalog) [reply packages/console/src/components/common/FileUpload/FileUpload.tsx]: Yes, the accepted values of files can be passed via `accepted` parameter and the help text below can be passed via `helpText` prop

**Review by ursucarina — APPROVED** [review]

**Review by jsonporter — APPROVED** [review]
